packages/rrweb/src/record/mutation.ts

```diff
@@ -438,10 +438,26 @@ export default class MutationBuffer {
         // text mutation's id was not in the mirror map means the target node has been removed
         .filter((text) => this.mirror.has(text.id)),
       attributes: this.attributes
-        .map((attribute) => ({
-          id: this.mirror.getId(attribute.node),
-          attributes: attribute.attributes,
-        }))
+        .map((attribute) => {
+          const attributes = attribute.attributes;
+          if (typeof attributes.style === 'string') {
+            const somAsStr = JSON.stringify(attribute.styleOM);
+            const unchangedAsStr = JSON.stringify(attribute.styleOMUnchanged);
+            if (
+              somAsStr.length < attributes.style.length &&
+              (somAsStr + unchangedAsStr).split('var(').length ===
+                attributes.style.split('var(').length
+            ) {
+              // use the compact style mutation format #464
+              // (CSSOM fails badly when var() is present on shorthand properties)
```

Comment:
I was a little confused, think this might clarify things for future me
```suggestion
            // check if compact style mutation is actually shorter than string style mutation
            // CSSOM fails badly when var() is present on shorthand properties, check if this is the case here
            if (
              somAsStr.length < attributes.style.length &&
              (somAsStr + unchangedAsStr).split('var(').length ===
                attributes.style.split('var(').length
            ) {
              // use the compact style mutation format #464
```